core/node/provider.go

```diff
@@ -0,0 +1,72 @@
+package node
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/ipfs/go-ipfs-config"
+	"github.com/ipfs/go-ipld-format"
+	"github.com/libp2p/go-libp2p-routing"
+	"go.uber.org/fx"
+
+	"github.com/ipfs/go-ipfs/exchange/reprovide"
+	"github.com/ipfs/go-ipfs/pin"
+	"github.com/ipfs/go-ipfs/provider"
+	"github.com/ipfs/go-ipfs/repo"
+)
+
+const kReprovideFrequency = time.Hour * 12
```

Comment:
don't use leading k in Go names; const kReprovideFrequency should be reprovideFrequency